kratos/utilities/normal_calculation_utils.cpp

```diff
@@ -22,6 +23,264 @@
+        VariableUtils().SetFlag(VISITED, false, rModelPart.Nodes());
+
+        for(auto& r_cond : rModelPart.Conditions()) {
+            for(auto& r_node: r_cond.GetGeometry()) {
+                r_node.Set(VISITED, true);
+            }
+        }
+
+        rModelPart.GetCommunicator().SynchronizeOrNodalFlags(VISITED);
+
+        VariableUtils().SetVariable(NORMAL, zero, rModelPart.Nodes(), VISITED);
+    } else {
+        // In serial iteratre normally over the condition nodes
+        for(auto& r_cond: rModelPart.Conditions()) {
+            for(auto& r_node: r_cond.GetGeometry()) {
+                noalias(r_node.FastGetSolutionStepValue(NORMAL)) = zero;
+            }
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void NormalCalculationUtils::InitializeNormals<Element>(ModelPart& rModelPart)
+{
+    // Resetting the normals
+    const array_1d<double,3> zero = ZeroVector(3);
+
+    if (rModelPart.GetCommunicator().GetDataCommunicator().IsDistributed()) {
+        // If Parallel make sure normals are reset in all partitions
+        VariableUtils().SetFlag(VISITED, false, rModelPart.Nodes());
+
+        for(auto& r_cond : rModelPart.Elements()) {
+            for(auto& r_node: r_cond.GetGeometry()) {
+                r_node.Set(VISITED, true);
+            }
+        }
+
+        rModelPart.GetCommunicator().SynchronizeOrNodalFlags(VISITED);
+
+        VariableUtils().SetVariable(NORMAL, zero, rModelPart.Nodes(), VISITED);
+    } else {
+        // In serial iteratre normally over the element nodes
+        for(auto& r_cond: rModelPart.Elements()) {
+            for(auto& r_node: r_cond.GetGeometry()) {
+                noalias(r_node.FastGetSolutionStepValue(NORMAL)) = zero;
+            }
+        }
+    }
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void NormalCalculationUtils::CalculateNormals<Condition>(
+    ModelPart& rModelPart,
+    const bool EnforceGenericAlgorithm
+    )
+{
+    // Getting process info
+    const auto& r_process_info = rModelPart.GetProcessInfo();
+
+    // Getting dimension
+    const SizeType dimension = r_process_info.GetValue(DOMAIN_SIZE);
+
+    // Sum all the nodes normals
+    auto& r_conditions_array = rModelPart.Conditions();
+    const auto it_cond_begin = r_conditions_array.begin();
+
+    // Checking if we can compute with simplex
+    const GeometryData::KratosGeometryType geometry_type = it_cond_begin->GetGeometry().GetGeometryType();
+    const bool use_simplex = EnforceGenericAlgorithm ? false : dimension == 2 ? geometry_type == GeometryData::KratosGeometryType::Kratos_Line2D2 : geometry_type == GeometryData::KratosGeometryType::Kratos_Triangle3D3;
```

Comment:
This feels like too much to grasp from one line.  Also, this seems to be the heart of the PR, so probably it would be good to have it in the following form:
```
    if (EnforceGenericAlgorithm)
    {
        use_simplex = false;
    }
    else if (dimension == 2)
    {
        use_simplex = (geomtype == line2d2);
    }
    else
    {
        use_simplex = (geomtype == triangle3d3);
    }
```